lib/PuppeteerSharp/Launcher.cs

```diff
@@ -127,16 +145,66 @@ public class Launcher
                 var keepAliveInterval = options.KeepAliveInterval;
 
                 _connection = await Connection.Create(browserWSEndpoint, connectionDelay, keepAliveInterval);
+
+                if (options.LogProcess)
+                {
+                    Console.WriteLine($"PROCESS COUNT: {Interlocked.Increment(ref _processCount)}");
+                }
+
                 return await Browser.CreateAsync(_connection, options, KillChrome);
             }
             catch (Exception ex)
             {
-                await ForceKillChrome();
+                ForceKillChrome();
                 throw new Exception("Failed to create connection", ex);
             }
 
         }
 
+        public static async Task TryDeleteDirectory(string folder, int times = 10, TimeSpan? delay = null)
+        {
+            if (times <= 0)
```

Comment:
maybe add a validation that `IsChromeClosed` should be `true`?

Reply:
I like it.